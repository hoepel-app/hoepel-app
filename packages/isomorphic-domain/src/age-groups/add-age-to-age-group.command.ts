import { CommandMetadata, CommandBase } from '@hoepel.app/ddd-library'

type Payload = {
  ageGroupName: string
  age: number
}

export class AddAgeToAgeGroupCommand extends CommandBase<Payload> {
  name = 'add-age-to-age-group-command' as const

  static create(
    ageGroupName: string,
    age: number,
    commandMetadata: CommandMetadata
  ): AddAgeToAgeGroupCommand {
    return new AddAgeToAgeGroupCommand(
      {
        age,
        ageGroupName,
      },
      commandMetadata
    )
  }

  get age(): number {
    return this.payload.age
  }

  get ageGroupName(): string {
    return this.payload.ageGroupName
  }
}
