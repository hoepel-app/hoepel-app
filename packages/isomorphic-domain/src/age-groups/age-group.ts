import { DayDate } from '@hoepel.app/types'
import {
  differenceInCalendarYears,
  differenceInDays,
  setYear,
  getYear,
} from 'date-fns'

export type SwitchOverOn = 'childs-birthday' | 'new-school-year'

export type AgeGroupProps = {
  name: string
  validForAges: readonly number[]
}

export class AgeGroup {
  private constructor(private readonly props: AgeGroupProps) {}

  belongsToThisGroup(
    birthDate: DayDate,
    switchOverOn: SwitchOverOn,
    currentDate: DayDate = DayDate.today()
  ): boolean {
    if (switchOverOn === 'childs-birthday') {
      const currentAge = differenceInCalendarYears(
        currentDate.nativeDate,
        birthDate.nativeDate
      )

      return this.validForAges.has(currentAge)
    } else {
      const newSchoolYear = this.dateIsInNewSchoolYear(currentDate)
      const hasHadBirthDay = this.hasHadBirthDayThisYear(birthDate, currentDate)

      const currentAge = differenceInCalendarYears(
        currentDate.nativeDate,
        birthDate.nativeDate
      )

      if (!newSchoolYear && hasHadBirthDay) {
        return this.validForAges.has(currentAge - 1)
      } else {
        return this.validForAges.has(currentAge)
      }
    }
  }

  static create(name: string, validForAges: ReadonlySet<number>): AgeGroup {
    return AgeGroup.fromProps({
      name,
      validForAges: [...validForAges],
    })
  }

  static fromProps(props: AgeGroupProps): AgeGroup {
    return new AgeGroup(props)
  }

  toProps(): AgeGroupProps {
    return this.props
  }

  withAgeAdded(age: number): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      validForAges: [...new Set([...this.validForAges, age])],
    })
  }

  withAgeRemoved(age: number): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      validForAges: [...this.validForAges].filter((a) => a !== age),
    })
  }

  withName(name: string): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      name,
    })
  }

  agesOverlapWith(other: AgeGroup): ReadonlySet<number> {
    const otherAges = [...other.validForAges]

    return new Set(
      [...this.validForAges].filter((age) => otherAges.includes(age))
    )
  }

  private hasHadBirthDayThisYear(
    birthDate: DayDate,
    currentDate: DayDate
  ): boolean {
    const daysSinceBirthday = differenceInDays(
      currentDate.nativeDate,
      setYear(birthDate.nativeDate, getYear(currentDate.nativeDate))
    )

    return daysSinceBirthday >= 0
  }

  private dateIsInNewSchoolYear(date: DayDate): boolean {
    return date.month >= 9
  }

  get name(): string {
    return this.props.name
  }

  get validForAges(): Set<number> {
    return new Set(this.props.validForAges)
  }
}
