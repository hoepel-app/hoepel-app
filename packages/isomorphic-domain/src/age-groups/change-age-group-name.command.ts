import { Command } from '../command/command'

export type ChangeAgeGroupNameCommandProps = {
  tenantId: string
  ageGroupCurrentName: string
  ageGroupNewName: string
}

export class ChangeAgeGroupNameCommand
  implements Command<ChangeAgeGroupNameCommandProps> {
  name = 'change-age-group-name-command' as const

  private constructor(private readonly props: ChangeAgeGroupNameCommandProps) {}

  static create(
    tenantId: string,
    ageGroupCurrentName: string,
    ageGroupNewName: string
  ): ChangeAgeGroupNameCommand {
    return new ChangeAgeGroupNameCommand({
      tenantId,
      ageGroupCurrentName,
      ageGroupNewName,
    })
  }

  toProps(): ChangeAgeGroupNameCommandProps {
    return this.props
  }

  get tenantId(): string {
    return this.props.tenantId
  }

  get ageGroupCurrentName(): string {
    return this.props.ageGroupCurrentName
  }

  get ageGroupNewName(): string {
    return this.props.ageGroupNewName
  }
}
