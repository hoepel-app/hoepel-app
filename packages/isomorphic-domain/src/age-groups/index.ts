export { AgeGroupsApplicationService } from './age-groups.application-service'
export { ChangeSwitchOverOnCommand } from './change-switch-over-on.command'
export { RemoveAgeGroupCommand } from './remove-age-group.command'
export { AddAgeGroupCommand } from './add-age-group.command'
export { AgeGroup, SwitchOverOn } from './age-group'
export { AgeGroups, AgeGroupsProps } from './age-groups'
export { AgeGroupsRepository } from './age-groups.repository'
