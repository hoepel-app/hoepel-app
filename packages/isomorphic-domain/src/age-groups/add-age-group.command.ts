import { CommandBase, CommandMetadata } from '@hoepel.app/ddd-library'
import { AgeGroupProps, AgeGroup } from './age-group'

type Payload = { ageGroupProps: AgeGroupProps }

export class AddAgeGroupCommand extends CommandBase<Payload> {
  name = 'add-age-group-command' as const

  static create(
    ageGroup: AgeGroup,
    metadata: CommandMetadata
  ): AddAgeGroupCommand {
    return new AddAgeGroupCommand(
      { ageGroupProps: ageGroup.toProps() },
      metadata
    )
  }

  get ageGroup(): AgeGroup {
    return AgeGroup.fromProps(this.payload.ageGroupProps)
  }
}
