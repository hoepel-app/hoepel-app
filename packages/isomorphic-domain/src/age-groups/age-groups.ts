import { AgeGroupProps, AgeGroup, SwitchOverOn } from './age-group'
import { DayDate } from '@hoepel.app/types'

export type AgeGroupsProps = {
  ageGroups: readonly AgeGroupProps[]
  switchOverOn: SwitchOverOn
}
export class AgeGroups {
  private constructor(private readonly props: AgeGroupsProps) {}

  get ageGroups(): readonly AgeGroup[] {
    return this.props.ageGroups.map((ageGroup) => AgeGroup.fromProps(ageGroup))
  }

  get names(): ReadonlySet<string> {
    return new Set(this.ageGroups.map((group) => group.name))
  }

  get switchOverOn(): SwitchOverOn {
    return this.props.switchOverOn
  }

  /** Ages that are not in use yet by any age group */
  get unusedAges(): ReadonlySet<number> {
    const usedAges = this.usedAges
    return new Set(
      [...AgeGroups.validAges.values()].filter((age) => !usedAges.has(age))
    )
  }

  get usedAges(): ReadonlySet<number> {
    const ages = this.ageGroups.map((group) => group.validForAges)

    return ages.reduce(
      (acc, curr) => new Set([...acc, ...curr]),
      new Set<number>()
    )
  }

  static fromProps(props: AgeGroupsProps): AgeGroups {
    return new AgeGroups(props)
  }

  static create(switchOverOn: SwitchOverOn): AgeGroups {
    return this.fromProps({
      switchOverOn,
      ageGroups: [],
    })
  }

  static createEmpty(): AgeGroups {
    return this.create('childs-birthday')
  }

  static get validAges(): ReadonlySet<number> {
    return new Set([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16])
  }

  toProps(): AgeGroupsProps {
    return {
      switchOverOn: this.switchOverOn,
      ageGroups: this.ageGroups.map((group) => group.toProps()),
    }
  }

  withAddedAgeGroup(ageGroup: AgeGroup): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      ageGroups: [...this.toProps().ageGroups, ageGroup.toProps()],
    })
  }

  withSwitchOverOn(switchOverOn: SwitchOverOn): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      switchOverOn,
    })
  }

  withAgeGroupRemoved(ageGroupName: string): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      ageGroups: this.ageGroups
        .filter((group) => group.name !== ageGroupName)
        .map((group) => group.toProps()),
    })
  }

  classifyPerson(
    birthDate: DayDate,
    currentDate: DayDate = DayDate.today()
  ): AgeGroup | null {
    return (
      this.ageGroups.find((ageGroup) =>
        ageGroup.belongsToThisGroup(birthDate, this.switchOverOn, currentDate)
      ) || null
    )
  }

  mayAddAgeGroup(ageGroup: AgeGroup): boolean {
    const overlap = this.ageGroups
      .map((group) => group.agesOverlapWith(ageGroup))
      .reduce((acc, set) => new Set([...acc, ...set]), new Set())

    const nameAlreadyExists = this.names.has(ageGroup.name)

    return overlap.size === 0 && !nameAlreadyExists
  }

  findAgeGroupWithName(name: string): AgeGroup | null {
    return this.ageGroups.find((group) => group.name === name) || null
  }
}
