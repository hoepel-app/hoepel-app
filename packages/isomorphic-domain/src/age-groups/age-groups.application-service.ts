import { AgeGroupsRepository } from './age-groups.repository'
import { CommandResult } from '@hoepel.app/ddd-library'
import { AddAgeGroupCommand } from './add-age-group.command'
import { ChangeSwitchOverOnCommand } from './change-switch-over-on.command'
import { RemoveAgeGroupCommand } from './remove-age-group.command'
import { AddAgeToAgeGroupCommand } from './add-age-to-age-group.command'
import { AgeGroups } from './age-groups'
import { Observable } from 'rxjs'
import { first } from 'rxjs/operators'
import { RemoveAgeFromAgeGroupCommand } from './remove-age-from-age-group.command'
import { ChangeAgeGroupNameCommand } from './change-age-group-name.command'

export class AgeGroupsApplicationService {
  constructor(private readonly repo: AgeGroupsRepository) {}

  findAgeGroups(tenantId: string): Observable<AgeGroups> {
    return this.repo.getForTenant(tenantId)
  }

  async addAgeGroup(command: AddAgeGroupCommand): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    if (!ageGroups.mayAddAgeGroup(command.ageGroup)) {
      return { status: 'rejected' }
    }

    const newAgeGroups = ageGroups.withAddedAgeGroup(command.ageGroup)

    await this.repo.put(newAgeGroups)

    return { status: 'accepted' }
  }

  async changeSwitchOverOn(
    command: ChangeSwitchOverOnCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(ageGroups.withSwitchOverOn(command.switchOverOn))

    return { status: 'accepted' }
  }

  async removeAgeGroup(command: RemoveAgeGroupCommand): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(ageGroups.withAgeGroupRemoved(command.ageGroupName))

    return { status: 'accepted' }
  }

  async removeAgeFromAgeGroup(
    command: RemoveAgeFromAgeGroupCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeRemovedFromAgeGroup(command.ageGroupName, command.age)
    )

    return { status: 'accepted' }
  }

  async addAgeToAgeGroup(
    command: AddAgeToAgeGroupCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeAddedToAgeGroup(command.ageGroupName, command.age)
    )

    return { status: 'accepted' }
  }

  async changeAgeGroupName(
    command: ChangeAgeGroupNameCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeGroupRenamed(
        command.ageGroupCurrentName,
        command.ageGroupNewName
      )
    )

    return { status: 'accepted' }
  }
}
