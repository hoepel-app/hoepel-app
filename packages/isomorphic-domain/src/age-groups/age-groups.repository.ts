import { AgeGroups } from './age-groups'
import { Observable } from 'rxjs'

export type AgeGroupsRepository = {
  getForTenant(tenantId: string): Observable<AgeGroups>
  put(entity: AgeGroups): Promise<void>
}
