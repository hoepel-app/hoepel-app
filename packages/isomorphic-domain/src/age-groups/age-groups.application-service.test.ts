import { AgeGroupsApplicationService } from './age-groups.application-service'
import { AgeGroups } from './age-groups'
import { AddAgeGroupCommand } from './commands/add-age-group.command'
import { AgeGroup } from './age-group'
import { ChangeSwitchOverOnCommand } from './commands/change-switch-over-on.command'
import { RemoveAgeGroupCommand } from './commands/remove-age-group.command'
import { of } from 'rxjs'
import { first } from 'rxjs/operators'
import { RemoveAgeFromAgeGroupCommand } from './commands/remove-age-from-age-group.command'
import { AddAgeToAgeGroupCommand } from './commands/add-age-to-age-group.command'
import { ChangeAgeGroupNameCommand } from './commands/change-age-group-name.command'
import { CommandMetadata } from '@hoepel.app/ddd-library'
import '@hoepel.app/ddd-library-test-utils'

describe('AgeGroupsApplicationService', () => {
  const exampleGroups = (tenantId: string): AgeGroups =>
    AgeGroups.create(tenantId, 'new-school-year')
      .withAddedAgeGroup(AgeGroup.create('Kleuters', new Set([2, 3, 4])))
      .withAddedAgeGroup(AgeGroup.create('Mini', new Set([5, 6, 7, 8])))
      .withAddedAgeGroup(AgeGroup.create('Maxi', new Set([9, 10])))
      .withAddedAgeGroup(AgeGroup.create('Tieners', new Set([11, 12, 13])))

  const commandMetadata: CommandMetadata = {
    commandId: 'my-command-id-123',
    requestedBy: {
      email: 'test@example.org',
      type: 'user',
      uid: 'my-uid-123',
    },
    tenantId: 'some-tenant-id',
    timestamp: new Date('2020-04-25T19:43:29.161Z'),
  }

  describe('addAgeGroup', () => {
    it('rejects if age group with name already exists', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }
      const service = new AgeGroupsApplicationService(repo)

      const command = AddAgeGroupCommand.create(
        AgeGroup.create('Tieners', new Set([12])),
        commandMetadata
      )

      const commandResult = await service.addAgeGroup(command)

      expect(commandResult).toBeRejected()
      expect(repo.put).not.toHaveBeenCalled()
    })

    it('accepts if age group can be added', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const command = AddAgeGroupCommand.create(
        AgeGroup.create('Oudere tieners', new Set([14])),
        commandMetadata
      )

      const commandResult = await service.addAgeGroup(command)

      expect(commandResult).toBeAccepted()
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })

  describe('changeSwitchOverOn', () => {
    it('changes switchover of age groups', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }
      const service = new AgeGroupsApplicationService(repo)

      const command = ChangeSwitchOverOnCommand.create(
        'childs-birthday',
        commandMetadata
      )

      const commandResult = await service.changeSwitchOverOn(command)

      expect(commandResult).toBeAccepted()
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })

  describe('removeAgeGroup', () => {
    it('removes an age group by name', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const command = RemoveAgeGroupCommand.create('Tieners', commandMetadata)

      const commandResult = await service.removeAgeGroup(command)

      expect(commandResult).toBeAccepted()
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })

  describe('findAgeGroups', () => {
    it('find age groups for a tenant', async () => {
      const groups = exampleGroups('my-tenant')

      const repo = {
        getForTenant: jest.fn(() => of(groups)),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const result = await service
        .findAgeGroups('my-tenant-name')
        .pipe(first())
        .toPromise()

      expect(result).toEqual(groups)
      expect(repo.getForTenant).toHaveBeenCalledTimes(1)
      expect(repo.getForTenant.mock.calls[0]).toEqual(['my-tenant-name'])
    })
  })

  describe('removeAgeFromAgeGroup', () => {
    it('removes age from age group and persists', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const group = await service.removeAgeFromAgeGroup(
        RemoveAgeFromAgeGroupCommand.create('Tieners', 12, commandMetadata)
      )

      expect(group).toBeAccepted()
      expect(repo.getForTenant).toHaveBeenCalledTimes(1)
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })

  describe('addAgeToAgeGroup', () => {
    it('adds age to age group and persists', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const group = await service.addAgeToAgeGroup(
        AddAgeToAgeGroupCommand.create('Tieners', 14, commandMetadata)
      )

      expect(group).toBeAccepted()
      expect(repo.getForTenant).toHaveBeenCalledTimes(1)
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })

  describe('changeAgeGroupName', () => {
    it('changes the name of an age group and persists', async () => {
      const repo = {
        getForTenant: jest.fn((tenantId: string) =>
          of(exampleGroups(tenantId))
        ),
        put: jest.fn(() => Promise.resolve()),
      }

      const service = new AgeGroupsApplicationService(repo)

      const group = await service.changeAgeGroupName(
        ChangeAgeGroupNameCommand.create(
          'Tieners',
          'Nieuwe Naam',
          commandMetadata
        )
      )

      expect(group).toBeAccepted()
      expect(repo.getForTenant).toHaveBeenCalledTimes(1)
      expect(repo.put).toHaveBeenCalledTimes(1)
      expect(repo.put.mock.calls[0]).toMatchSnapshot()
    })
  })
})
