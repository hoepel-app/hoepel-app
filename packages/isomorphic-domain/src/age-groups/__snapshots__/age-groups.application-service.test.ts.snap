// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeGroupsApplicationService addAgeGroup accepts if age group can be added 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
        Object {
          "name": "Tieners",
          "validForAges": Array [
            11,
            12,
            13,
          ],
        },
        Object {
          "name": "Oudere tieners",
          "validForAges": Array [
            14,
          ],
        },
      ],
      "switchOverOn": "new-school-year",
      "tenantId": "some-tenant-id",
    },
  },
]
`;

exports[`AgeGroupsApplicationService addAgeToAgeGroup adds age to age group and persists 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
        Object {
          "name": "Tieners",
          "validForAges": Array [
            11,
            12,
            13,
            14,
          ],
        },
      ],
      "switchOverOn": "new-school-year",
      "tenantId": "some-tenant-id",
    },
  },
]
`;

exports[`AgeGroupsApplicationService changeAgeGroupName changes the name of an age group and persists 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
        Object {
          "name": "Nieuwe Naam",
          "validForAges": Array [
            11,
            12,
            13,
          ],
        },
      ],
      "switchOverOn": "new-school-year",
      "tenantId": "some-tenant-id",
    },
  },
]
`;

exports[`AgeGroupsApplicationService changeSwitchOverOn changes switchover of age groups 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
        Object {
          "name": "Tieners",
          "validForAges": Array [
            11,
            12,
            13,
          ],
        },
      ],
      "switchOverOn": "childs-birthday",
      "tenantId": "some-tenant-id",
    },
  },
]
`;

exports[`AgeGroupsApplicationService removeAgeFromAgeGroup removes age from age group and persists 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
        Object {
          "name": "Tieners",
          "validForAges": Array [
            11,
            13,
          ],
        },
      ],
      "switchOverOn": "new-school-year",
      "tenantId": "some-tenant-id",
    },
  },
]
`;

exports[`AgeGroupsApplicationService removeAgeGroup removes an age group by name 1`] = `
Array [
  AgeGroups {
    "props": Object {
      "ageGroups": Array [
        Object {
          "name": "Kleuters",
          "validForAges": Array [
            2,
            3,
            4,
          ],
        },
        Object {
          "name": "Mini",
          "validForAges": Array [
            5,
            6,
            7,
            8,
          ],
        },
        Object {
          "name": "Maxi",
          "validForAges": Array [
            9,
            10,
          ],
        },
      ],
      "switchOverOn": "new-school-year",
      "tenantId": "some-tenant-id",
    },
  },
]
`;
